examples/mlmc.cpp

```diff
@@ -209,14 +216,32 @@ int main(int argc, char* argv[])
     // Construct "finite volume mass" matrix using mfem
     mfem::RT_FECollection sigmafec(0, nDimensions);
     mfem::ParFiniteElementSpace sigmafespace(pmesh, &sigmafec);
-
     {
         mfem::ParBilinearForm a(&sigmafespace);
         a.AddDomainIntegrator(
             new FiniteVolumeMassIntegrator(*spe10problem.GetKInv()) );
-        a.Assemble();
-        a.Finalize();
-        a.SpMat().GetDiag(weight);
+
+        if (elem_mass == false)
+        {
+            a.Assemble();
+            a.Finalize();
+            a.SpMat().GetDiag(weight);
+        }
+        else
+        {
+            local_weight.resize(pmesh->GetNE());
+            mfem::DenseMatrix M_el_i;
+            for (unsigned int i = 0; i < local_weight.size(); i++)
+            {
+                a.ComputeElementMatrix(i, M_el_i);
+                mfem::Vector& local_weight_i = local_weight[i];
+                local_weight_i.SetSize(M_el_i.Height());
+                for (int j = 0; j < local_weight_i.Size(); j++)
+                {
+                    local_weight_i[j] = 1.0 / M_el_i(j, j);
+                }
+            }
+        }
     }
 
     for (int i = 0; i < weight.Size(); ++i)
```

Comment:
This loop should also probably be moved into the `if (elem_mass == false)` block.

Reply:
Good catch!